lumen/ai/assistant.py

```diff
@@ -226,9 +226,14 @@ async def run_demo(event):
 
     def _add_analysis_suggestions(self):
         pipeline = memory['current_pipeline']
-        applicable_analyses = [
-            analysis for analysis in self._analyses if analysis.applies(pipeline)
-        ]
+        current_analysis = memory.get("current_analysis")
+        applicable_analyses = []
+        for analysis in self._analyses:
+            if analysis.applies(pipeline):
+                if current_analysis and not current_analysis._consecutive_calls:
+                    if analysis.__name__ == current_analysis.__name__:
+                        continue
+                applicable_analyses.append(analysis)
```

Comment:
Why compare on `__name__` instead of identity? Also those nested conditionals aren't great, would rather `continue` early.

```suggestion
        current_analysis = memory.get("current_analysis")
        allow_consecutive = getattr(current_analysis, '_consecutive_calls', True)
        applicable_analyses = []
        for analysis in self._analyses:
            if analysis.applies(pipeline) and (allow_consecutive or analysis is not current_analysis):
                applicable_analyses.append(analysis)
```

Reply:
I did not test this ☝️ to be clear.

Reply:
It was close :) missing `type`

<img width="680" alt="image" src="https://github.com/user-attachments/assets/a7858f42-290f-43b8-aea9-3205eac4017a">